src/build-tools/near-bindgen-exporter.js

```diff
@@ -1,69 +1,243 @@
-                  t.callExpression(t.memberExpression(classId, t.identifier('_reconstruct')), [t.identifier('_contract'), t.identifier('_state')]))])) : t.emptyStatement(),
-              // let _args = Counter._getArgs();
-              t.variableDeclaration('let', [t.variableDeclarator(t.identifier('_args'), t.callExpression(t.memberExpression(classId, t.identifier('_getArgs')), []))]),
-              // let _result = _contract.method(args);
-              t.variableDeclaration('let', [t.variableDeclarator(t.identifier('_result'), t.callExpression(t.memberExpression(t.identifier('_contract'), t.identifier(method)), [t.identifier('_args')]))]),
-              contractMethods[method] === 'initialize' || contractMethods[method] === 'call' ?
-                // Counter._saveToStorage(_contract);
-                t.expressionStatement(t.callExpression(t.memberExpression(classId, t.identifier('_saveToStorage')), [t.identifier('_contract')]))
-                : t.emptyStatement(),
-              // if (_result !== undefined) 
-              //   if (_result && _result.constructor && _result.constructor.name === 'NearPromise')
-              //     _result.onReturn();
-              //   else
-              //     near.valueReturn(_contract._serialize(result));
-              t.ifStatement(t.binaryExpression('!==', t.identifier('_result'), t.identifier('undefined')), 
-                t.ifStatement(t.logicalExpression('&&', t.logicalExpression('&&', t.identifier('_result'), t.memberExpression(t.identifier('_result'), t.identifier('constructor'))), 
-                                                        t.binaryExpression('===', t.memberExpression(t.memberExpression(t.identifier('_result'), t.identifier('constructor')), t.identifier('name')), t.stringLiteral('NearPromise'))),
-                  t.expressionStatement(t.callExpression(t.memberExpression(t.identifier('_result'), t.identifier('onReturn')), [])),
-                  t.expressionStatement(t.callExpression(t.memberExpression(t.identifier('env'), t.identifier('value_return')), [t.callExpression(t.memberExpression(classId, t.identifier('_serialize')), [t.identifier('_result')])]))))]))));
-            console.log(`Babel ${method} method export done`);
+
+          for (let methodName of Object.keys(contractMethods)) {
+            path.insertAfter(
+              t.exportNamedDeclaration(
+                t.functionDeclaration(
+                  t.identifier(methodName),
+                  [],
+                  t.blockStatement([
+                    // Read the state of the contract from storage.
+                    // const _state = Counter._getState();
+                    readState(classId),
+                    // Throw if initialized on any subsequent init function calls.
+                    // if (_state) { throw new Error('Contract already initialized'); }
+                    preventDoubleInit(contractMethods[methodName]),
```

Comment:
These helper code generators are really cool! Easier to read then our previous version. cc @volovyks